### PR TITLE
Use map.style.getImage() instead of internal maplibre variables

### PR DESCRIPTION
**Pull request by ZeLonewolf**
Based on initial discussions in maplibre/maplibre-gl-js#2159, this PR changes the sprite sheet access from the internal variable map.style.imageManager.images[id] to map.style.getImage(id). There is no map.getImage(id) function at the top level, so this is the most top-level function we can call to extract images.

### CODE REVIEW

**Comment by ZeLonewolf** [comment]
This PR is blocked by maplibre/maplibre-gl-js#2164. It cannot be merged until we can utilize this fix.

**Comment by ZeLonewolf** [comment]
Ready for review.  This change will allow us to be compatible with maplibre 3.0.0.  After that cutover, we'll want to change the access method to `Map.getImage()` as implemented in maplibre/maplibre-gl-js#786

File: src/js/shield.js [hunk 1]
```diff
@@ -377,11 +377,23 @@ export function romanizeRef(ref) {
   return roman + ref.slice(number.toString().length);
 }
 
-export function generateShieldCtx(map, id) {
-  return generateSpriteCtx(map.style.imageManager.images, id);
+export function getDrawnShieldBounds(id) {
+  let routeDef = getRouteDef(id);
+  let shieldDef = getShieldDef(routeDef);
```
Comment (1ec5) [inline src/js/shield.js:382]: Both of this function’s call sites already have `shieldDef` and `routeDef` in context that they can pass in. That would make it a little easier to unit test this function.

Comment (ZeLonewolf) [reply src/js/shield.js]: Good call, fixed in bd85e51

**Review by 1ec5 — APPROVED** [review]

**Pull request merged**